### PR TITLE
#180 | Transaction row

### LINKED ISSUE #180: transaction row component
- create externallink component to show abbreviated address which links to teloscan, as well as more info link
    - should support addresses like 0x1234...5678
    - should support plain text like "More info"
    - show external link icon
- create timestamp component
    - given unix time, display text like "5 days ago"
    - on hover, show tooltip with actual date like "Feb 12, 2023 12:09:18 PM (UTC -05:00)"

### PR DESCRIPTION
**Pull request by EJSG**
# Fixes #180 

## Description
Adds a wallet transaction row component

this PR also updates:
- wallet balance row component - change breakpoints to small rather than medium, update tooltip display logic
    - note that this PR does not fully solve https://github.com/telosnetwork/telos-wallet/issues/258, as it would conflict with Viter's upcoming PR to update when numbers are truncated 
- tooltip component - add slot and ability to hide icon, fix mobile show/hide behavior

## Test scenarios
- go to https://deploy-preview-260--wallet-staging.netlify.app/
- sign in with metamask

**Wallet balance row & tooltip changes**
- go to https://deploy-preview-260--wallet-staging.netlify.app/evm/wallet?tab=balance
- use dev tools to make screen mobile with touch
    - truncated numbers with a fiat value should no longer have a tooltip icon next to them
- tap on a truncated primary (large, bold) balance
    - full balance should show in a tooltip
- tap on a truncated secondary (small, gray) balance 
    - full balance should show

**Transaction row**
- go to https://deploy-preview-260--wallet-staging.netlify.app/evm/wallet?tab=transactions
- you should see a list of transactions in various states
- compare rows to transaction rows [in the design](https://www.figma.com/file/eIpbnt6vqwqptEbZWtwO8Y/v0.3?node-id=131-658&t=IioT2DO6jt6JBLim-4) at both mobile and desktop screens
    - designs should match rows
- click the links on rows
    - they should go to the right place, either address or tx route of telsocan
- hover the timestamp with your mouse / tap on mobile
    - full date should show in a tooltip
- values without a fiat price should show a tooltip next to them. click that tool tip
    - should contain info that no fiat price is available

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file
-   [x] I have created a new issue with the required translations for the currently supported languages
-   [x] I have added appropriate test coverage

### CODE REVIEW

File: src/components/ToolTip.vue [hunk 1]
```diff
@@ -9,6 +9,7 @@ const warningIcon = require('src/assets/icon--warning.svg');
 const props = defineProps<{
     text?: string,
```
Comment (pmjanus) [inline src/components/ToolTip.vue:10]: required

File: src/pages/evm/wallet/WalletBalanceRow.vue [hunk 2]
```diff
@@ -264,12 +242,12 @@ export default defineComponent({
         },
     },
     methods: {
-        toggleTooltip() {
-            this.showTooltip = true;
+        // used only to format the balance shown in tooltips, the rest of formatting is handled in computed props
+        formatBalance(amount: number, isFiat: boolean) {
```
Comment (pmjanus) [inline src/pages/evm/wallet/WalletBalanceRow.vue:246]: here maybe we can name function  formatTooltipBalance and remove comment

File: src/pages/evm/wallet/WalletTransactionsTab.vue [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+export default defineComponent({
+    name: 'WalletTransactionsTab',
+    components: {
+        WalletTransactionRow,
+    },
+    computed: {
+        shapedTransactions(): ShapedTransactionRow[] {
+            return [{
+                id: '0x'.concat('1'.repeat(40)),
```
Comment (pmjanus) [inline src/pages/evm/wallet/WalletTransactionsTab.vue:18]: I'm thinking maybe this could be an import and used on the tests as wells.

**Review by pmjanus — APPROVED** [review]
Nice job, left some comments, none of them block merging imo.

**Comment by Viterbo** [comment]
@karynemayer and @EJSG  I love it !! :heart: 

image

File: src/components/TimeStamp.vue [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (donnyquixotic) [reply src/components/TimeStamp.vue:3]: I think it's fine to leave it for a quick follow up task. https://github.com/telosnetwork/telos-wallet/issues/267

File: src/antelope/stores/utils/index.ts [hunk 5]
```diff
@@ -171,6 +171,18 @@ export function getFormattedUtcOffset(date: Date): string {
...
 }
 
+/**
+ * Given a unix timestamp, returns a date in the form of Jan 1, 2023 07:45:22 AM
+ *
+ * @param epoch
+ *
+ * @return string
+ */
+export function getLongDate(epoch: number): string {
```
[the remark this answers is not in the document]
Comment (donnyquixotic) [reply src/antelope/stores/utils/index.ts:181]: Yes, in WalletTransactionRow, i'm updating to use the new lib now.

**Review by donnyquixotic — APPROVED** [review]

File: src/components/ExternalLink.vue [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+const props = defineProps<{
+    text: string,
+    url: string,
```
[the remark this answers is not in the document]
Comment (EJSG) [reply src/components/ExternalLink.vue:6]: Yes. Both are needed to make a link

File: src/components/ToolTip.vue (hunk 1 above)
Comment (EJSG) [reply src/components/ToolTip.vue:10]: Doesn't need to be required, it is intentionally optional. You can have a warning without also having non warning text

**Pull request merged**